program_intent_engagement/apps/api/v1/views.py

```diff
@@ -1 +1,61 @@
-# Create your views here.
+"""
+V1 API Views
+"""
+from rest_framework import status
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from edx_rest_framework_extensions.auth.jwt.authentication import JwtAuthentication
+from django.db.models import F, Window
+from django.db.models.functions import RowNumber
+
+from program_intent_engagement.apps.core.models import ProgramIntent
+from program_intent_engagement.apps.api.serializers import ProgramIntentSerializer
+
+
+
+
+# TODO Do I add an @schema thing here?
+class MostRecentAndCertainIntentsView(APIView):
+    """
+    Retrieve a list of user intents
+    This endpoint returns a list of intents specified by the user.
+      Grouped by programs and sorted by most recent CERTAIN intent
+    Path: /api/[version]/intents/most-recent-and-certain
+    Returns:
+     * 200: OK
+     * 400: Query returned no results
+    """
+    authentication_classes = (JwtAuthentication,)
+    serializer_class = ProgramIntentSerializer
+
+    def get_queryset(self):
+        """
+        This view should return the best intents per program
+        for the currently authenticated user.
+
+        Best intent is defined as the most recent CERTAIN intent
+        """
+        user = self.request.user
+
+        result = (ProgramIntent.objects
+                      .filter(user=user)
+                      .order_by(F('effective_date'), F('certainty'))
```

Comment:
You don't need to use F objects here. You can just reference the fields by name. I don't think you're getting any advantage by using an F object here because the order by clause isn't going to evaluate and hit the database until the queryset itself is evaluated. This same is true for the F object in the annotate clause.

```suggestion
                      .order_by('effective_date', 'certainty')
```

Reply:
Hm ok, I read about them [here](https://docs.djangoproject.com/en/4.0/ref/models/expressions/#f-expressions), but don't completely understand their use case.